docs/encoders.md

```diff
@@ -39,3 +43,112 @@ That's it, now on every reboot the motor will turn in one direction until it fin
+* `<axis>.encoder.config.use_index = True`
+* `odrv0.save_configuration()`
+* `odrv0.reboot()`
+
+and now see if after a reboot you can run: 
+* `<axis>.requested_state = AXIS_STATE_ENCODER_INDEX_SEARCH`
+
+without getting errors. 
+
+## Encoder Problems
+
+There are several issues that may prevent you from completing encoder calibration. 
+
+ODrive may not complete the calibrate sequence when you go to:
+* `<axis>.requested_state = AXIS_STATE_FULL_CALIBRATION_SEQUENCE`
+
+It completes the calibrate sequence after:
+* `<axis>.requested_state = AXIS_STATE_FULL_CALIBRATION_SEQUENCE`
+
+but fails after you go to:
+* `odrvN.axisN.requested_state = AXIS_STATE_CLOSED_LOOP_CONTROL`
+
+Or maybe it vibrates in an entertaining way. See:
+https://www.youtube.com/watch?v=gaRUmwvSyAs
+
+## Encoder Signals
+
+If your encoder is properly connected, run the command:
+* `<axis>.encoder.shadow_count `
+
+and look at your value. Then turn your motor by hand and see if that value changes. Also, notice that the command:
+* `<axis>.encoder.config.cpr = 4000`
+
+must reflect the number of counts odrive receives after one complete turn of the motor. So use shadow_count to test if that is working properly. 
+
+You will probably never be able to properly debug if you have problems unless you use an oscilloscope. If you have one, try the following:
+Connect to the AB pins, see if you get square waves as you turn the motor.
+Connect to the I pin, see if you get a pulse on a complete rotation. Sometimes this is hard to see.
+If you are using SPI, connect to CLK, and CS pins, look for a signal like this:
+[INSERT PIC]
+
+## Encoder Noise
+Noise is found in all circuits, life is just about figuring out if it is preventing your system from working. Lots of users have no problems with noise interferring with their odrive operation, others will tell you "_I've been using the same encoder as you with no problems_". Power to 'em, that may be true, but it doesn't mean it will work for you. If you are concerned about noise, there are several possible sources:
+
+* Encoder wires may be too close to motor wires
+* Long wires between encoder and ODrive
+* Use of ribbon cable
+
+The following _might_ mitigate noise problems. Use twisted pairs, where one side of each twisted pair is tied to ground, the other side is tied to your signal. If you are using SPI, use a 20-50 ohm resistor in series on CLK, which is more susceptable noise.
+
+## AS5047/AS5048 Encoders
+The AS5047/AS5048 encoders are Hall Effect/Magenetic sensors that can serve as rotary encoders for the ODrive.
+
+The AS5047 has 3 independent output interfaces: SPI, ABI, and PWM. 
+The AS5048 has 4 independent output interfaces: SPI, ABI, I2C, and PWM.
+
+Both chips come with evaluation boards that can simplify mounted the chips to your motor. For our purposes if you are using an evaluation board you should select the settings for 3.2v, and tie MOSI high to 3.2v. 
```

Comment:
3.2v? pretty sure the rail on the odrive is 3.3

Reply:
corrected